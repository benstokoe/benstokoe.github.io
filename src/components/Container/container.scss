@import '../../scss/variables';
@import '../../scss/spacing';

.container {
  margin: 0 auto;
  max-width: $site-width;
  padding: spacing(3) spacing(3);

  &:after {
    content: '';
    display: table;
    clear: both;
  }
}
