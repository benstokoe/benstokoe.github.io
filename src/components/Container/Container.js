import React from 'react';
import PropTypes from 'prop-types';
import './container.scss';

const Container = ({ children }) => <div className="container">{children}</div>;

Container.propTypes = {
  children: PropTypes.node,
};

export default Container;
