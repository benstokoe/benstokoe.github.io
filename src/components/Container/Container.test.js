import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container', () => {
  let wrapper;
  const props = {
    children: <p>Children</p>,
  };

  beforeEach(() => {
    wrapper = shallow(<Container {...props} />);
  });

  it('should render properly with no props', () => {
    wrapper = shallow(<Container />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
