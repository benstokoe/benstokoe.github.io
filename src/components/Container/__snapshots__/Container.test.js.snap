// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Container should render properly 1'] = '
<div
  className="container"
>
  <p>
    Children
  </p>
</div>
';

exports['Container should render properly with no props 1'] = '
<div
  className="container"
/>
';
